spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarConfiguration.java

```diff
@@ -176,4 +179,12 @@ PulsarFunctionAdministration pulsarFunctionAdministration(PulsarAdministration p
 				properties.isFailFast(), properties.isPropagateFailures(), properties.isPropagateStopFailures());
 	}
 
+	@Bean
+	@Scope("prototype")
```

Comment:
Why is this a prototype bean? Do the factories mutate the builder and need to do so independently?

Reply:
The factories only use the stateless `getFullyQualifiedName` method on the builder currently. However, this bean will also be used when users want to create `PulsarTopic` beans though. In that case, the builder has "name" and "numberOfPartitions" state that would vary during building.

The other alternative is to provide a "start building" method on the builder which would clone the builder. 

```java
public PulsarTopicBuilder forName(String name) {
    return this.cloneBuilder();
}
```

I saw other builders using prototype in SB so I went w/ that approach. I meant to call this point out during my self-review of the PR. 

Any preference?

Reply:
> However, this bean will also be used when users want to create PulsarTopic beans though

Thanks for the explanation. This sounds like a good reason for it to be a prototype.